src/string/isomorphism.rs

```diff
@@ -0,0 +1,56 @@
+/*
+Given two strings s and t, determine whether they are isomorphic.
+The two strings are isomorphic if the characters in s can be replaced by some mapping relation to get t
+*/
+use std::collections::HashMap;
+
+pub fn is_isomorphic(s: &str, t: &str) -> bool {
+    if s.len() != t.len() {
+        return false;
+    }
+    let sv: Vec<char> = s.chars().collect();
+    let tv: Vec<char> = t.chars().collect();
+    let mut sr1 = String::new();
+    let mut map1 = HashMap::new();
+    for i in 0..sv.len() {
+        if let Some(x) = map1.get(&sv[i]) {
+            sr1.push(*x);
+        } else {
+            map1.insert(sv[i], tv[i]);
+            sr1.push(tv[i]);
+        }
+    }
+    let mut sr2 = String::new();
+    let mut map2 = HashMap::new();
+    for i in 0..sv.len() {
+        if let Some(x) = map2.get(&tv[i]) {
+            sr2.push(*x);
+        } else {
+            map2.insert(tv[i], sv[i]);
+            sr2.push(sv[i]);
+        }
+    }
+    sr1 == t && sr2 == s
+}
```

Comment:
```suggestion
pub fn is_isomorphic(s: &str, t: &str) -> bool {
    let s_chars: Vec<char> = s.chars().collect();
    let t_chars: Vec<char> = t.chars().collect();
    if s_chars.len() != t_chars.len() {
        return false;
    }
    let mut s_to_t_map = HashMap::new();
    let mut t_to_s_map = HashMap::new();

    for (s_char, t_char) in s_chars.into_iter().zip(t_chars) {
        if !check_mapping(&mut s_to_t_map, s_char, t_char)
            || !check_mapping(&mut t_to_s_map, t_char, s_char)
        {
            return false;
        }
    }

    true
}

fn check_mapping(map: &mut HashMap<char, char>, key: char, value: char) -> bool {
    match map.get(&key) {
        Some(&mapped_char) => mapped_char == value,
        None => {
            map.insert(key, value);
            true
        }
    }
}
```

Please note that there are few things in this suggestion:
- the _bug_ when two strings have same number of characters but different number of bytes,
- avoids need of constructing additional strings,
- reduced the code duplication.

Reply:
ok